inductiva/tests/mixins/test_file_manager.py

```diff
@@ -0,0 +1,201 @@
+    """
+
+    mixins.file_manager.render_file(source_file, target_file, **render_args)
+
+    with open(target_file, "r", encoding="utf-8") as file:
+        file_content = file.read()
+
+    assert expected_output in file_content
+
+
+def test_set_root_dir__valid_input__creates_folder():
+    """Check that with a non-empty input name the root dir is created.
+    
+    Goal: Test root dir creation from a given name. The folder is
+    created in the current working dir, so we want to verify that the
+    root dir was actually created, that the name is correct and that it
+    is instantiated in the current working_dir.
+    """
+
+    root_dir = "test_root_dir"
+    file_manager = mixins.FileManager()
+
+    file_manager.set_root_dir(root_dir)
+    created_root_dir = file_manager.get_root_dir()
+    assert os.path.isdir(created_root_dir)
+    assert root_dir in created_root_dir.name
+    assert pathlib.Path(os.getcwd()) == created_root_dir.parent
+
+
+def test_set_root_dir_null_input__raises_error():
+    """Check that with a null input name an error is raised.
+    
+    Goal: Test the usability of the set_root_dir method when root_dir is None.
+    We capture the error that we raise (ValueError) and match the sentence to
+    validate it is the correct error.
+    """
+    file_manager = mixins.FileManager()
+    with pytest.raises(ValueError) as excinfo:
+        file_manager.set_root_dir(None)
+
+    assert str(excinfo.value) == "Given root directory cannot be None"
+
+
+def test_get_root_dir__root_is_none__create_default_dir():
+    """Check that with a null root dir the default dir is created.
+    
+    Goal: Test the scenario when the root_dir is created by default, when
+    no root_dir is set first. Hence, the workflow should be still valid to
+    not raise errors during the workflow. 
+    """
+    file_manager = mixins.FileManager()
+    root_dir = file_manager.get_root_dir()
+
+    assert os.path.isdir(root_dir)
+    assert root_dir.name == file_manager.DEFAULT_ROOT_DIR
+
+
+@pytest.mark.parametrize(
+    "source_file, target_file",
+    [(os.path.join(TEST_DIR, "test_file.txt"), None),
+     (os.path.join(TEST_DIR, "test_file.txt"), "added_file.txt")])
+def test_add_file__valid_file__to_target(source_file, target_file):
+    """Check that a non-empty input file is added to the given target.
+    
+    Goal: Test on of the functionalities of the FileManager().add_file method,
+    which adds a source file into a target file, without rendering any args.
+    In this case, the add_file function just copies the file directly.
+    Hence, we verify if the target_file exists after being copied.
+    """
+
+    file_manager = mixins.FileManager()
+    target_file = file_manager.add_file(source_file, target_file)
+
+    root_dir = file_manager.get_root_dir()
+    assert os.path.isfile(os.path.join(root_dir, target_file))
+
+
+@pytest.mark.parametrize("source_file, render_args, target_file",
```

Comment:
to prevent very large indentantions, why not import `parametrize` independently?
We have very small chances for name clashing and the function name is clearly understandable in the test context.

Reply:
The bare minimum is from pytest import mark (exactly the same workflow if we don't set `from . import mixins` in the init.